File: problems/Kth Smallest Element in a BST/kthSmallest.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int KthSmallest(TreeNode root, int k) {
        var nums = inorder(root, new List<int>());
        return nums[k - 1];
    }

    public List<int> inorder(TreeNode root, List<int> arr) {
        if (root == null) return arr;
        inorder(root.left, arr);
        arr.Add(root.val);
        inorder(root.right, arr);
        return arr;
    }
}

File: problems/Kth Smallest Element in a BST/kthSmallest.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int kthSmallest(TreeNode root, int k) {
        Stack<TreeNode> bfs = new Stack<>();

        while (true) {
            while (null != root) {
                bfs.push(root);
                root = root.left;
            }

            root = bfs.pop();

            if (0 == --k) {
                return root.val;
            }

            root = root.right;
        }
    }
}
